### PR TITLE
pino v7 transport

### PR DESCRIPTION
**Pull request by Eomm**
Ref https://github.com/pinojs/pino/issues/1056

Adding support for pino transport v7: https://github.com/pinojs/pino/blob/master/docs/transports.md

I'm trying to keep the pino legacy transport support: is this needed or this PR may support new transport interface only?

This PR is still draft, but feel free to suggest tests or anything else 👍🏽

### CODE REVIEW

**Comment by Eomm** [comment]
cc @mcollina @davidmarkclements

**Comment by mcollina** [comment]
> I'm trying to keep the pino legacy transport support: is this needed or this PR may support new transport interface only?

It is needed.

**Comment by mcollina** [comment]
Good work! What is missing? A lot of tests are failing.

File: psock.js [hunk 1]
```diff
@@ -10,10 +10,11 @@ const through2 = require('through2')
 const nopt = require('nopt')
 const fs = require('fs')
 
-// TODO quick and dirty
 if (require.main !== module) {
+  // used as module
   module.exports = transport
 } else {
+  // usaed as cli
```
Comment (mcollina) [inline psock.js:17]: You can always export and just call the cli if it's a main.

**Review by mcollina — APPROVED** [review]
lgtm

File: Readme.md [hunk 2]
```diff
@@ -18,8 +18,30 @@ $ npm install --production -g pino-socket
...
+  options: {
+    address: '10.10.10.5',
+    port: 5000.
+    mode: 'tcp'
+  }
+})
+pino(transport)
+```
+
+All the options are described further down.
```
Comment (jsumners) [inline Readme.md:38]:
```suggestion
All options are described further below.
```

File: psock.js (hunk 1 above)
Comment (jsumners) [reply psock.js:16]:
```suggestion
  // used as cli
```

**Review by jsumners — APPROVED** [review]

**Comment by Eomm** [comment]
As a warning, the CI did not run

> 1 workflow awaiting approval

**Pull request merged**